packages/react-native-renderer/src/__tests__/ReactNativeMount-test.internal.js

```diff
@@ -249,6 +249,106 @@ describe('ReactNative', () => {
     });
   });
 
+  it('should call UIManager.measure on ref.measure', () => {
+    const View = createReactNativeComponentClass('RCTView', () => ({
+      validAttributes: {foo: true},
+      uiViewClassName: 'RCTView',
+    }));
+
+    class Subclass extends ReactNative.NativeComponent {
+      render() {
+        return <View>{this.props.children}</View>;
+      }
+    }
+
+    const CreateClass = createReactClass({
+      mixins: [NativeMethodsMixin],
+      render() {
+        return <View>{this.props.children}</View>;
+      },
+    });
+
+    [View, Subclass, CreateClass].forEach(Component => {
+      UIManager.measure.mockClear();
+
+      let viewRef;
+      ReactNative.render(
+        <Component
+          ref={ref => {
+            viewRef = ref;
+          }}
+        />,
+        11,
+      );
+
+      expect(UIManager.measure).not.toBeCalled();
+
+      const successCallback = jest.fn();
+      viewRef.measure(successCallback);
+
+      expect(UIManager.measure).toHaveBeenCalledTimes(1);
+      expect(UIManager.measure).toHaveBeenCalledWith(
+        expect.any(Number),
+        expect.any(Function),
+      );
+
+      const args = UIManager.measure.mock.calls[0];
+      expect(successCallback).not.toBeCalled();
+      args[1]('success');
+      expect(successCallback).toBeCalledWith('success');
```

Comment:
This seems a little janky too.

Reply:
Here, I agree with you. The coupling is tight as a side effect of an array being used for the arguments instead of a hash with a named key. Given the design of the code, I’m not sure what else to do except make a test helper function that wraps this detail and give it an intention-revealing name.

Reply:
The specific reason I wanted to test this is because one of the UIManager measure calls takes the arguments `fail, success`, but the host component takes the arguments `success, fail` and flips the order when passed to the UIManager. I wanted to validate that this stays the correct order because it is trivially easy to break (or think that the existing code is wrong).

I can't assert that the callback *is* the callback that is passed in because it is wrapped, so calling it and validating the argument felt like the most appropriate thing.